# Group Member Names:
### Claire Jones, Samuel Harris, Henry Jones, Ryan Jamison, Nashrullah Safdari, Eric Hoerdermann

# Initial Analysis
### Part 1 of the lab 1 instructions

In [20]:
import pandas as pd
import numpy as np

In [21]:
# 1.1 (Load the foreign gifts data)
gift = pd.read_csv('ForeignGifts_edu.csv')

In [22]:
# 1.2 (Describe the differences between different classes of gift types)
gift['Gift Type'].unique()

array(['Monetary Gift', 'Contract', 'Real Estate'], dtype=object)

There are 3 types of gifts: Monetary, Contract, and Real Estate

<div class="alert alert-block alert-warning">
<b>Question 1:</b>  Which country gives the most money, in total?
</div>

In [23]:
# 1.3.1
gift.groupby('Country of Giftor')['Foreign Gift Amount'].sum().sort_values(ascending=False).head()

Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
Name: Foreign Gift Amount, dtype: int64

Qatar gives the most total money, at $2,706,240,869

<div class="alert alert-block alert-warning">
<b>Question 2:</b>  Which country initiates the most gifts, by count?
</div>

In [24]:
# 1.3.2
gift['Country of Giftor'].value_counts().head()

Country of Giftor
ENGLAND        3655
CHINA          2461
CANADA         2344
JAPAN          1896
SWITZERLAND    1676
Name: count, dtype: int64

England initiates the most gifts by count (3655 individual gifts).

<div class="alert alert-block alert-warning">
<b>Question 3:</b>  Which country gives the largest gifts, on average?
</div>

In [25]:
# 1.3.3
gift.groupby('Country of Giftor')['Foreign Gift Amount'].mean().sort_values(ascending=False).head()

Country of Giftor
BERMUDA    7.688837e+06
ECUADOR    3.985140e+06
QATAR      3.905109e+06
ANTIGUA    3.435545e+06
JERSEY     3.064589e+06
Name: Foreign Gift Amount, dtype: float64

Bermuda gives the largest gifts on average (approx. $7.7m on average)

<div class="alert alert-block alert-warning">
<b>Question 4:</b>  Which institution receive the most money, in total?
</div>

In [26]:
# 1.3.4
gift.groupby('Institution Name')['Foreign Gift Amount'].sum().sort_values(ascending=False).head()

Institution Name
Carnegie Mellon University               1477922504
Cornell University                       1289937761
Harvard University                        954803610
Massachusetts Institute of Technology     859071692
Yale University                           613441311
Name: Foreign Gift Amount, dtype: int64

Carnagie Mellon University recieves the most money in total

<div class="alert alert-block alert-warning">
<b>Question 5:</b>  Which institution receive the greatest number of gifts, by count?
</div>

In [27]:
# 1.3.5
gift['Institution Name'].value_counts().head()


Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Ohio State University (The)                    1014
Johns Hopkins University                        899
Name: count, dtype: int64

The University of California, Los Angelas recieves the most gifts by count.

<div class="alert alert-block alert-warning">
<b>Question 6:</b>  What is the average amount each university receives? Median amount? If they are different, why?
</div>

In [28]:
# 1.3.6
uni_totals = gift.groupby('Institution Name')['Foreign Gift Amount'].sum()

mean_total_per_uni = uni_totals.mean()
median_total_per_uni = uni_totals.median()

print(f"Mean donation total by Uni: {mean_total_per_uni:,.2f}")
print(f"Median donation total by Uni: {median_total_per_uni:,.2f}")

Mean donation total by Uni: 52,202,878.97
Median donation total by Uni: 6,486,791.50


The difference between the mean and median donations to universities is due to the distribution being skewed heavily by uncommon, large donations. This pulls the mean amount higher much faster than it does the median amount. 

<div class="alert alert-block alert-warning">
<b>Question 7:</b>  What is the largest flow of a single country to a single institution?
</div>

In [29]:
# 1.3.7
gift.groupby(['Country of Giftor', 'Institution Name'])['Foreign Gift Amount'].sum().reset_index().sort_values('Foreign Gift Amount', ascending=False).head()

,Country of Giftor,Institution Name,Foreign Gift Amount
1865,QATAR,Cornell University,1018473315
159,BERMUDA,Carnegie Mellon University,750000000
1884,QATAR,Texas A&M University,504099350
1864,QATAR,Carnegie Mellon University,425410702
1872,QATAR,Georgetown University,327128310


The largest flow of a single country to a single insitution is Qatar to Cornell. Qatar has donated $1,018,473,315 to Cornell.

<div class="alert alert-block alert-warning">
<b>Question 8:</b>  Which are the top giftors to US academic institutions?
</div>

In [30]:
# 1.3.8
gift.groupby('Giftor Name')['Foreign Gift Amount'].agg(Total_Amount='sum', Gift_Count='count').sort_values('Total_Amount', ascending=False).head(10)

# VSCode shows and overload warning / error for this line, but seems to run fine anyways.

,Total_Amount,Gift_Count
Giftor Name,,
Qatar Foundation,1166503744,261
Qatar Foundation/Qatar National Res,796197000,10
Qatar Foundation for Education,373945215,6
Anonymous,338793629,262
Saudi Arabian Cultural Mission,275221475,480
HCL,190000000,5
Church of Jesus Christ of LDS,185203715,7
Emirates Institute for Advanced Sc,170641244,24
QIC,148355497,2


The table above shows the top 10 giftors to US academic institutions. There are many funding sources from middle eastern countries, and some anonymous sources as well.